base/sparse/cholmod.jl

```diff
@@ -42,16 +42,28 @@ const common_postorder = (1:4) + cholmod_com_offsets[17]
 ## macro to generate the name of the C function according to the integer type
 macro cholmod_name(nm,typ) string("cholmod_", eval(typ) == SuiteSparse_long ? "l_" : "", nm) end
 
-for Ti in IndexTypes
-    @eval begin
-        function common(::Type{$Ti})
-            a = fill(0xff, common_size)
-            @isok ccall((@cholmod_name "start" $Ti
-                , :libcholmod), Cint, (Ptr{UInt8},), a)
-            set_print_level(a, 0) # no printing from CHOLMOD by default
-            a
-        end
+function start(a::Vector{UInt8})
+    @isok ccall((@cholmod_name("start", SuiteSparse_long), :libcholmod),
+        Cint, (Ptr{UInt8},), a)
+end
+
+function finish(a::Vector{UInt8})
+    @isok ccall((@cholmod_name("finish", SuiteSparse_long), :libcholmod),
+        Cint, (Ptr{UInt8},), a)
+end
+
+function defaults(a::Vector{UInt8})
+    @isok ccall((@cholmod_name("defaults", SuiteSparse_long), :libcholmod),
+        Cint, (Ptr{UInt8},), a)
+end
+
+function common(default = false)
```

Comment:
@tkelman This signature now has a boolean argument which defaults to `false`. Only when it's `true` the global state is set to the default values and I only use `true` in `cholfact` and `ldltfact`.